src/javascript/app/pages/trade/charts/highchart.js

```diff
@@ -598,7 +594,7 @@ const Highchart = (() => {
     };
 
     const setStopStreaming = () => {
-        if (chart && (is_sold || is_settleable)) {
+        if (chart && (is_sold || is_settleable) && (isSoldBeforeExpiry() ? sell_time : end_time)) {
```

Comment:
`(isSoldBeforeExpiry() ? sell_time : end_time)` check ensures that sell or end line is drawn